### PR TITLE
dApp: Your balance - tooltip for in progress deposits

### PR DESCRIPTION
**Pull request by kpyszkowski**
Closes: AENG-47

### Changes
- `useActivities` hook: added status whether at least one of activity is pending
- `PositionDetails` component: depending on the activities status conditionally rendered the tooltip
- `Tooltip` component: added `wrapperProps` prop to provide more control over styling of the component

### CODE REVIEW

File: dapp/src/hooks/store/useActivities.ts [hunk 1]
```diff
@@ -2,5 +2,12 @@ import { selectActivities } from "#/store/wallet"
 import { useAppSelector } from "./useAppSelector"
 
 export default function useActivities() {
-  return useAppSelector(selectActivities)
+  const data = useAppSelector(selectActivities)
+  const status: "pending" | "idle" = data.some(
+    (activity) => activity.status === "pending",
+  )
+    ? "pending"
+    : "idle"
```
Comment (kkosiorowska) [inline dapp/src/hooks/store/useActivities.ts:10]: The term `status` is a bit confusing to me. What I mean is that an activity can have a `status`, here I understand it more as a fetching `status`. Maybe we should change the name to `hasAnyPendingActivity` and return boolean value.


Additionally, I think we can create a separate selector for this data. The [role of selectors](https://redux.js.org/usage/deriving-data-selectors#basic-selector-concepts) is to return data based on that state. So it seems that we can keep this approach.

```
export const selectHasAnyPendingActivity = createSelector(
  (state: RootState) => state.wallet.activities,
  (activities) => activities.some((activity) => activity.status === "pending"),
)
```

Comment (kpyszkowski) [reply dapp/src/hooks/store/useActivities.ts]: Ref commit: 18a4238c

File: dapp/src/pages/DashboardPage/PositionDetails.tsx [hunk 2]
```diff
@@ -51,12 +58,29 @@ export default function PositionDetails() {
...
+              label="Your balance will update once the pending deposit is finalized."
+              placement="right"
+              wrapperProps={{
+                lineHeight: 4,
+              }}
+            >
+              <Icon
+                as={IconClockHour5Filled}
+                color={TOOLTIP_ICON_COLOR}
+                opacity={0.25}
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/PositionDetails.tsx]: Ref commit: 2f40fe53

File: dapp/src/pages/DashboardPage/PositionDetails.tsx [hunk 3]
```diff
@@ -51,12 +58,29 @@ export default function PositionDetails() {
...
-          Your Acre balance
+        <HStack>
+          <TextMd>Your Acre balance</TextMd>
+          {activities.status === "pending" && (
+            <Tooltip
+              label="Your balance will update once the pending deposit is finalized."
+              placement="right"
+              wrapperProps={{
+                lineHeight: 4,
+              }}
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/PositionDetails.tsx]: Ref commit: 1e2d62f5

File: dapp/src/pages/DashboardPage/PositionDetails.tsx [hunk 4]
```diff
@@ -22,6 +24,8 @@ import UserDataSkeleton from "#/components/shared/UserDataSkeleton"
 import { featureFlags } from "#/constants"
 import { TextMd } from "#/components/shared/Typography"
 import AcreTVLMessage from "./AcreTVLMessage"
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/PositionDetails.tsx]: Ref commit: 36051ba1

File: dapp/src/pages/DashboardPage/PositionDetails.tsx [hunk 5]
```diff
@@ -51,12 +54,21 @@ export default function PositionDetails() {
...
+
   return (
     <Flex w="100%" flexDirection="column" gap={5}>
       <VStack alignItems="start" spacing={0}>
         {/* TODO: Component should be moved to `CardHeader` */}
-        <TextMd>
-          Your Acre balance
+        <HStack>
+          <TextMd>Your Acre balance</TextMd>
+          {!hasPendingActivities && (
```
[the remark this answers is not in the document]
Comment (kpyszkowski) [reply dapp/src/pages/DashboardPage/PositionDetails.tsx]: My bad, left for debugging
528be241

File: dapp/src/store/wallet/walletSelector.ts [hunk 6]
```diff
@@ -13,6 +13,11 @@ export const selectActivities = createSelector(
   (activities) => sortActivitiesByTimestamp(activities),
 )
 
+export const selectHasPendingActivities = createSelector(
+  (state: RootState) => state.wallet.activities,
+  (activities) => activities.some((activity) => activity.status === "pending"),
+)
+
```
Comment (kkosiorowska) [inline dapp/src/store/wallet/walletSelector.ts:20]: I have confirmed the requirements and the tooltip should only show up when the user has pending deposits.

```suggestion
export const selectHasPendingDeposits = createSelector(
  (state: RootState) => state.wallet.activities,
  (activities) =>
    activities.some(
      (activity) =>
        activity.status === "pending" && activity.type === "deposit",
    ),
)
```

**Review by kkosiorowska — APPROVED** [review]

**Pull request merged**